File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    n=[]
    for i in moves:
        for j in board:
            if j[i-1] == 0:continue
            else:
                n.append(j[i-1])
                j[i-1]=0
                break
    i=0
    answer=0
    while i < len(n)-1:
        if n[i]==n[i+1]:
            del n[i]
            del n[i]
            answer+=2
            i=0
            continue
        i+=1
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    const storage = [];
    
    for(let m of moves){
        const temp = m-1;
        
        for(let b of board){
            // console.log(b);
            if(b[temp]==0){
                continue;
            }else{
                storage.push(b[temp]);
                b[temp]=0;
                break;
            }
        }
        // console.log(storage);
        let flag = true;
        
        while(storage.length>=2 && storage[storage.length-2]===storage[storage.length-1]){
                    storage.pop();
                    storage.pop();
                    answer+=2;
        }
    }
    
    return answer;
}